### PR TITLE
Updates based on ABDK feedback

### PR DESCRIPTION
**Pull request by balasan**

### CODE REVIEW

File: src/core/CorePermissions.sol [hunk 1]
```diff
@@ -69,9 +69,13 @@ abstract contract CorePermissions is ICorePermissions, AccessControlEnumerableUp
         }
     }
 
-    function revokeRole(bytes32 role, address account) public override onlyRole(getRoleAdmin(role)) {
+    function revokeRole(bytes32 role, address account)
+        public
+        override(IAccessControlUpgradeable, AccessControlUpgradeable)
+        onlyRole(getRoleAdmin(role))
+    {
+        require(msg.sender != account, "NO_SELFREVOKE");
```
Comment (ttarsi) [inline src/core/CorePermissions.sol:77]: nice, can you add a comment in here as to why we req this? so that there is always at least 1 governor

**Review by ttarsi — COMMENTED** [review]
cool nice work on this so far

File: src/core/CorePermissions.sol (hunk 1 above)
Comment (balasan) [reply src/core/CorePermissions.sol:77]: actually - do we need limit this check to `GOVERN_ROLE`?
just realized it now applies to all the roles...

Comment (balasan) [reply src/core/CorePermissions.sol:78]: ah right, because other cannot self-revoke anyway

**Review by ttarsi — APPROVED** [review]
lgtm!!

**Pull request merged**